src/OAuth2/GrantType/Saml2Bearer.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace OAuth2\GrantType;
+
+use OAuth2\ClientAssertionType\ClientAssertionTypeInterface;
+use OAuth2\ResponseType\AccessTokenInterface;
+use OAuth2\RequestInterface;
+use OAuth2\ResponseInterface;
+use OAuth2\Saml\Saml2AssertionInterface;
+
+/**
+ * The Saml2 bearer authorization grant implements SAML2 assertions as a grant type per the IETF draft.
+ *
+ * NOTE: the authorization grant is not a clientAssertion.
+ * @see http://tools.ietf.org/html/draft-ietf-oauth-saml2-bearer-16#section-4
+ *
+ * @author aacotroneo
+ * @author Brent Shaffer <bshafs at gmail dot com>
+ */
+class Saml2Bearer implements GrantTypeInterface
+{
+
+    protected $samlAssertion;
+
+    /**
+     * @param array $settings configuration for the default SAML2Assertion
+     * @param Saml2AssertionInterface $samlAssertion
+     */
+    public function __construct($settings, Saml2AssertionInterface $samlAssertion = null)
```

Comment:
It's used in line 34 for the default case. It's not used when you provide a Saml2AssertionInterface indeed (I used that for testing). I don't know how that can be reflected better.